docs/2.guide/4.recipes/4.sessions-and-authentication.md

```diff
@@ -0,0 +1,403 @@
+        <button type="submit">Login</button>
+      </div>
+
+      <div v-if="error">{{ error }}</div>
+    </form>
+  </div>
+</template>
+
+```
+
+### 5. Create an API route for login
+
+With the login form created, we need to create an API route to handle the login request. This route should accept a POST request with the email and password in the request body and check the email and password against the database. If the user and password match, we'll set a session cookie to log the user in.
+
+This server API route should be at `/server/api/auth/login.post.ts`. Just like the registration form endpoint, suffixing the filename in `.post.ts.` means that this handler will only respond to post requests.
+
+```typescript [/server/api/auth/login.post.ts]
+import users from "~/database/schema/users";
+import getDatabase from "~/database/database";
+import { eq } from "drizzle-orm";
+import bcrypt from "bcrypt";
+
+export default defineEventHandler(async (event) => {
+  const db = await getDatabase();
+
+  const foundUser = (
+    await db
+      .select({ id: users.id, name: users.name, email: users.email, password: users.password })
+      .from(users)
+      .where(eq(users.email, email))
+      .limit(1)
+  )?.[0];
+
+  // compare the password hash
+  if (!foundUser || !bcrypt.compareSync(password, foundUser.password)) {
+    // return an error if the user is not found or the password doesn't match
+    throw createError({
+      statusCode: 401,
+      statusMessage: "Invalid email or password",
+    });
+  }
+
+  // log in as the selected user
+  await replaceUserSession(event, {
+    user: {
+      id: user.id,
+      name: user.name,
+    },
+    loggedInAt: new Date(),
+  });
+});
+
+
+```
+
+The user should now be logged in! With the session set, we can get the current user session in any API route or page by calling `getUserSession(event)` which is auto-imported as a util function from the `nuxt-auth-utils` package.
+
+### 6. Create a logout API route
+
+Users need to be able to log out, so we should create an API route to allow them to do this. This should require post request as well, just like login. We'll clear the session when this endpoint is called.
+
+We'll use the `clearUserSession` from the `auth-utils` module to log the user out and clear the session data. This function is automatically imported from the module by Nuxt, so we don't need to manually import it.
+
+```typescript [server/api/auth/logout.post.ts]
+export default defineEventHandler(async (event) => {
+  // Clear the current user session
+  await clearUserSession(event);
+});
+```
+
+### 7. Protect your server route
+
+Protecting server routes is key to making sure your data are safe. Front-end middleware is helpful for the user, but without back-end protection your data can still be accessed. Because of this, it is critical that we protect any API routes with sensitive data. For these sensitive routes, we should return a 401 error if the user is not logged in.
```

Comment:
data is safe (data is singular in this specific case)
Also, to use less technical terms, we might want to rephrase front-end and back-end into client-side and server-side.